packages/docusaurus/src/client/serverEntry.js

```diff
@@ -14,75 +14,74 @@ import {getBundles} from 'react-loadable-ssr-addon';
 import Loadable from 'react-loadable';
 
 import path from 'path';
-import fs from 'fs';
+import fs from 'fs-extra';
 import routes from '@generated/routes';
 import preload from './preload';
 import App from './App';
 import ssrTemplate from './templates/ssr.html.template';
 
 // Renderer for static-site-generator-webpack-plugin (async rendering via promises)
-export default function render(locals) {
+export default async function render(locals) {
   const {routesLocation} = locals;
   const location = routesLocation[locals.path];
-  return preload(routes, location).then(() => {
-    const modules = new Set();
-    const context = {};
-    const appHtml = ReactDOMServer.renderToString(
-      <Loadable.Capture report={moduleName => modules.add(moduleName)}>
-        <StaticRouter location={location} context={context}>
-          <App />
-        </StaticRouter>
-      </Loadable.Capture>,
-    );
+  await preload(routes, location);
+  const modules = new Set();
+  const context = {};
+  const appHtml = ReactDOMServer.renderToString(
+    <Loadable.Capture report={moduleName => modules.add(moduleName)}>
+      <StaticRouter location={location} context={context}>
+        <App />
+      </StaticRouter>
+    </Loadable.Capture>,
+  );
 
-    const helmet = Helmet.renderStatic();
-    const htmlAttributes = helmet.htmlAttributes.toString();
-    const bodyAttributes = helmet.bodyAttributes.toString();
-    const metaStrings = [
-      helmet.title.toString(),
-      helmet.meta.toString(),
-      helmet.link.toString(),
-    ];
-    const metaAttributes = metaStrings.filter(Boolean);
+  const helmet = Helmet.renderStatic();
+  const htmlAttributes = helmet.htmlAttributes.toString();
+  const bodyAttributes = helmet.bodyAttributes.toString();
+  const metaStrings = [
+    helmet.title.toString(),
+    helmet.meta.toString(),
+    helmet.link.toString(),
+  ];
+  const metaAttributes = metaStrings.filter(Boolean);
 
-    const {outDir} = locals;
-    const manifestPath = path.join(outDir, 'client-manifest.json');
-    const manifest = JSON.parse(fs.readFileSync(manifestPath, 'utf8'));
```

Comment:
Actually this PR started just to convert this readFileSync lol ....

Because each render function is called asynchronously and this opens up possibility of introducing `jest-worker` (or should we use native worker thread) to speed up html rendering build time